arch/arm/core/aarch64/vector_table.S

```diff
@@ -55,6 +55,15 @@ _ASM_FILE_PROLOGUE
 	mrs	\xreg1, elr_el1
 	stp	\xreg0, \xreg1, [sp, ___esf_t_spsr_elr_OFFSET]
 
+#ifdef CONFIG_USERSPACE
+	/* Save the current kernel/user mode in the context */
+	mrs	\xreg0, tpidrro_el0
+	str	\xreg0, [sp, ___esf_t_tpidrro_el0_OFFSET]
+
+	/* Switch TPIDRRO_EL0 to kernel mode */
+	msr	tpidrro_el0, xzr
```

Comment:
`tpidr_el0` is being used for thread local storage. Would there be any conflict?

Reply:
`tpidr_el0` and `tpidrro_el0` are two different registers. No conflict.